crates/symbolicator/src/services/symbolication/process_minidump.rs

```diff
@@ -250,18 +250,16 @@ impl SymbolicatorSymbolProvider {
 impl SymbolProvider for SymbolicatorSymbolProvider {
     async fn fill_symbol(
         &self,
-        module: &(dyn Module + Sync),
+        _module: &(dyn Module + Sync),
         _frame: &mut (dyn FrameSymbolizer + Send),
     ) -> Result<(), FillSymbolError> {
-        let debug_id = module.debug_identifier().ok_or(FillSymbolError {})?;
-
-        // Symbolicator's CFI caches never store symbolication information. However, we could hook
-        // up symbolic here to fill frame info right away. This requires a larger refactor of
-        // minidump processing and the types, however.
-        match self.files.contains_key(&debug_id) {
-            true => Ok(()),
-            false => Err(FillSymbolError {}),
-        }
+        // Always return an error here to signal that we have no useful symbol information to
+        // contribute. Doing nothing and reporting Ok trips a check in rust_minidump's
+        // instruction_seems_valid_by_symbols function that leads stack scanning to stop prematurely.
+        // See https://github.com/rust-minidump/rust-minidump/blob/7eed71e4075e0a81696ccc307d6ac68920de5db5/minidump-processor/src/stackwalker/mod.rs#L295.
+        //
+        // TODO: implement this properly, i.e., use symbolic to actually fill in information.
```

Comment:
Are we tracking this anywhere? Your change didn't introduce the need to implement all of this, but it would be good to create a ticket or issue somewhere (JIRA?) so we can easily track this work across projects. "Easily" in this case being subject to one's ability to use JIRA, ofc.

Reply:
I don’t think we ever want to implement this tbh. We have stackwalking as completely separate things, so I don’t see us merging that stuff. That would essentially mean loading/creating symcaches during stackwalking, and doing symcache lookups. We have a separate "stage" that does this.

Reply:
Should I still create a Jira ticket to keep an eye on this?

Reply:
> I don’t think we ever want to implement this tbh.

This does contradict what the TODO says so maybe we could just remove that entirely, in case somebody else stumbles upon this.

What I'm a little unclear on now is how this fits into the entire pipeline. Is this a situation where a different step (or "stage") in the pipeline fills in these symbols if needed, and the responsibility of this function should be strictly limited to streamlining address-checking? Is this actually a limitation that may lead to some symbols falling through the cracks? 

If we're potentially losing some data from the minidump because of what we're doing in this method, then a jira ticket (and keeping the TODO) makes sense to me. Otherwise we could just mention why we're doing what we're doing here in the context of the entire pipeline (see second paragraph about this method's responsibilities) and call it a day.

Reply:
Shoot, I missed that they check `has_name`. I think we should instead keep the logic here but write some placeholder function name in, like `<unknown>`.

At least when I ran my tests, this made a significant difference, because otherwise the stack walker would yield too many false-positive frames. Let's please test this change extensively.

Reply:
I'm also in favour of doing this TODO comment and tracking it somewhere.  We will get better stackwalking results from this.  Architecturally there isn't a lot of argument for this two stage thing with iterative stackwalking we have going on right now.

@jan-auer wrt filling out the name to `<unknown>` if you look at the code consuming this for the stackwalker this is identical to returning an `Err`, so there's no benefit to doing that.  I must admit I was also worried about this rollout and I thought a long time about whether we should have the whole partial rollout with diffing infrastructure again.  I was softly persuaded to not do so, but could easily be swayed back.

Reply:
@flub is right; in your proposed version, either we have the file, in which case we write a dummy function name and `instruction_seems_valid_by_symbols` accepts the address as valid, or we don't have the file, in which case we return an error and `instruction_seems_valid_by_symbols` accepts the address as valid.

Reply:
🙈 thanks for clarifying, somehow I read this branch completely wrong. Makes all sense now 👍